# Python SDK gen failing on relative tsp path

## question
This is my 1st time trying to create our Python SDK. The pipeline appears to be failing due to relative paths for included tsp files. Any idea how to get around this?

```
Errors occurred while generating SDK from specification/discoverydev/Discovery.Workspace/tspconfig.yaml. Follow the steps at https://aka.ms/azsdk/sdk-automation-faq#how-to-view-the-detailed-sdk-generation-errors to view detailed errors.
2025-10-08 16:27:29 [ERROR] ====== Error occurred in tsp compiler (error stack start) ======
2025-10-08 16:27:29 [ERROR] /mnt/vss/_work/1/s/azure-sdk-for-python-pr/sdk/discovery-workspace/azure-discovery-workspace/TempTypeSpecFiles/Discovery.Workspace/common.tsp:1:1 - error import-not-found: Couldn't resolve import "../Discovery.Shared/namespace.tsp"
2025-10-08 16:27:29 [ERROR] /mnt/vss/_work/1/s/azure-sdk-for-python-pr/sdk/discovery-workspace/azure-discovery-workspace/TempTypeSpecFiles/Discovery.Workspace/common.tsp:3:1 - error import-not-found: Couldn't resolve import "../Discovery.Shared/data-plane.tsp"
2025-10-08 16:27:29 [ERROR] /mnt/vss/_work/1/s/azure-sdk-for-python-pr/sdk/discovery-workspace/azure-discovery-workspace/TempTypeSpecFiles/Discovery.Workspace/common.tsp:4:1 - error import-not-found: Couldn't resolve import "../Discovery.Shared/shared.tsp"
2025-10-08 16:27:29 [ERROR] /mnt/vss/_work/1/s/azure-sdk-for-python-pr/sdk/discovery-workspace/azure-discovery-workspace/TempTypeSpecFiles/Discovery.Workspace/_project.tsp:5:1 - error import-not-found: Couldn't resolve import "../Discovery.Shared/data-plane.tsp"
2025-10-08 16:27:29 [ERROR] /mnt/vss/_work/1/s/azure-sdk-for-python-pr/sdk/discovery-workspace/azure-discovery-workspace/TempTypeSpecFiles/Discovery.Workspace/_conversation_message_logs.tsp:6:1 - error import-not-found: Couldn't resolve import "../Discovery.Shared/data-plane.tsp"
2025-10-08 16:27:29 [ERROR] /mnt/vss/_work/1/s/azure-sdk-for-python-pr/sdk/discovery-workspace/azure-discovery-workspace/TempTypeSpecFiles/Discovery.Workspace/_tool.tsp:5:1 - error import-not-found: Couldn't resolve import "../Discovery.Shared/data-plane.tsp"
2025-10-08 16:27:29 [ERROR] /mnt/vss/_work/1/s/azure-sdk-for-python-pr/sdk/discovery-workspace/azure-discovery-workspace/TempTypeSpecFiles/Discovery.Workspace/_tool.tsp:6:1 - error import-not-found: Couldn't resolve import "../Discovery.Shared/shared.tsp"
```
https://github.com/Azure/azure-rest-api-specs-pr/pull/24925/checks?check_run_id=52272214251

## answer
According to error log, it seems that your typespec refer to folder outside project folder but you don't declare it in `tspconfig.yaml`. You could add the missing dir like https://github.com/Azure/azure-rest-api-specs/blob/a0e813c3f64a7027fec1830ca35d11a78ddf19aa/specification/keyvault/Security.KeyVault.RBAC/tspconfig.yaml#L35-L37 then try again.

# Sphinx errors in generated code for Azure Batch

## question
I'm running into some errors in our `_operations` files but since this is generated code I'm not sure if they should be fixed manually.

After doing some investigation, there are some spacing inconsistencies in some of the functions in `_operations` but not sure why there would be discrepancies between the functions that are running into the issue and functions that aren't. Looking at the TypeSpec, they look like they're handled the same way. Like for example our `list-certificates` is missing a space in front of the URL while the issue doesn't exist in `list-pools`.

Link to PR: [\[Batch\] Beta Data Plane SDK 15.0.0b3 by cRui861 · Pull Request #42790 · Azure/azure-sdk-for-python](https://github.com/Azure/azure-sdk-for-python/pull/42790)
Linke to sphinx errors: https://dev.azure.com/azure-sdk/public/_build/results?buildId=5391082&view=logs&j=7b8caaf3-aec7-5f50-aa20-46d4a6342ac3&t=286a7ccb-1111-5827-fc90-5c974fed4202

## answer
I propose we temporarily reassign your TMPDIR to check if something got messed up along the way.
```
mkdir -p "$HOME/.tmp"
chmod 700 "$HOME/.tmp"
TMPDIR="$HOME/.tmp"
```

# Azure Analytics Experimentation unavailable for ExP customers

## question
This package should be generally available for FTE, but inaccessible both to ExP developers and customers:

I am still waiting on Internal permissions, but here is the link for the package which should be accessible: https://dev.azure.com/azure-sdk/internal/_artifacts/feed/azure-sdk-for-python-pr/PyPI/azure-analytics-experimentation/overview/1.0.0a20240625001

These links should also be visible for general FTE:
https://dev.azure.com/azure-sdk/internal/_artifacts/feed/azure-sdk-for-python-pr/connect/pip
https://aka.ms/exp/sdk

## answer
You only need read access to internal to gain access to the feed. Internal MS FTE will have access to that feed, but external access outside of FTE is not currently supported by these feeds. For those customers you will likely need to share the wheel or sdist yourself.

# Release Date Discrepancy: Kickoff vs. Release-Request

## question
Hi team,
I have a quick question regarding the October SDK release date.
The October SDK Release Kickoff notes mention that the all package ship date is 10/07, but in the release-request, the target release date is listed as "No later than 10/23/2025".
Just wanted to confirm — what’s the actual release date for our package? Are we still aiming for 10/07?
Thanks in advance!

## answer
For management plane, the guidance is here [Release SDK](https://eng.ms/docs/products/azure-developer-experience/develop/sdk-release/sdk-release#when-to-release).  If the service is using TypeSpec, then no bound with monthly train, you can take control of the TypeSpec to SDK release flow to release SDKs as long as service APIs and specs are publicly available.


# ARM Manifest Rollout May Be Delayed — How to Request SDK Release Exception?

## question
Hi team,
Due to potential delays in the ARM manifest rollout before 10/7, we’re considering requesting an exception to release our SDKs on 10/14. I’d like to clarify the process.
Here are the SDK details:
- Languages: Java / Python / JavaScript / Go
- Azure Product: HDInsight
- API Version: `2025-01-15-preview`
A few questions:
    1. What’s the process for requesting an exception?
    - Who should we email?
    2. Should the email content only include what was filled in during the release plan creation?
    - Is there any additional information we need to provide?
If anyone has a sample email or has gone through this process before, I’d really appreciate your input!

## answer
The wiki guidance for SDK exception is [Request an exception for Azure SDKs](https://eng.ms/docs/products/azure-developer-experience/onboard/request-exception)

# Release blocker. Getting issue with whl_no_aio in pipeline.

## question
Hello Language - Python
I am getting whl_no_aio: FAIL code

Could someone help?  There where only 4 PR since last releases.
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=5398611&view=logs&j=7a0a9bea-f59d-59d9-8e37-bf8eaa134735&t=8a5fc5c9-0246-59e1-4e9d-26ff73bc57ad&l=8197

## answer
Here is the issue: Pipelines - Run 20250929.8 logs

what `whl_no_aio` is doing is uninstalling `aiohttp` explicitly, and ensuring that your `sync` tests properly run without aiohttp installed.

it then runs your sync tests by setting additional pytest argument `--ignore-glob='*async*.py'`

you're seeing a failure because that test file doesn't have `async` in the name, but is attempting to import an async library -> explosion

1. update anything that actually imports `aio` or the like to exist in a test with `async` in the name
2. or skip `whl_no_aio` via `pyproject.toml` and run the risk of having an accidentall dep on `aiohttp` when you didn't intend to
